vowpalwabbit/parse_example.cc

```diff
@@ -76,9 +77,17 @@ class TC_parser
 
   inline void parserWarning(const char* message, char* begin, char* pos, const char* message2)
   {
-    cerr << message << std::string(begin, pos - begin).c_str() << message2 << "in Example #"
-         << this->p->end_parsed_examples << ": \"" << std::string(this->beginLine, this->endLine).c_str() << "\""
-         << endl;
+    if (p->strict_parse) {
+      std::stringstream exc_msg;
+      exc_msg << message << std::string(begin, pos - begin).c_str() << message2 << "in Example #"
+              << this->p->end_parsed_examples << ": \"" << std::string(this->beginLine, this->endLine).c_str() << "\""
+              << endl;
+      throw std::domain_error(exc_msg.str());
```

Comment:
I'd prefer this to be a VW specific exception of some type like `strict_parse_error`

Reply:
I've tried this in fede09b. Do you think it should be a subclass of `vw_exception`?

Reply:
Yep, in the same way the other exception types in vw_exception.h are